### PR TITLE
Adds documentation for Lupusec alarm control panel

### PR DESCRIPTION
**Pull request by majuss**
**Description:**
This PR add support for the alarm control panel from Lupus Electronics. For now the panel and sensors are supported and tested.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** https://github.com/home-assistant/home-assistant/pull/17691

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+```yaml
+# Example configuration.yaml entry
+lupusec:
+      username: user 
+      password: 'greatpassword'
+      ip_address: 192.168.178.35
+      scan_interval: 2
+```
+
+Configuration variables:
```
Comment (klaasnicolaas) [inline source/_components/alarm_control_panel.lupusec.markdown]: Use the new configuration variable syntax: [more info](https://developers.home-assistant.io/docs/en/documentation_create_page.html)

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+To use Lupusec devices in your installation, add the following `lupusec` section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+lupusec:
+      username: user 
+      password: 'greatpassword'
+      ip_address: 192.168.178.35
+      scan_interval: 2
```
Comment (frenck) [inline source/_components/alarm_control_panel.lupusec.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+The `lupus electronics` security control panel platform allows you to control your [Lupusec](https://www.lupus-electronics.de) devices.
+
+The requirement is that you have setup your [Lupusec component](/components/lupusec/).
+
+To use Lupusec devices in your installation, add the following `lupusec` section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+lupusec:
+      username: user
```
Comment (frenck) [inline source/_components/alarm_control_panel.lupusec.markdown]: Please use all CAPS constant like placeholders. For example, `YOUR_USERNAME`.

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+The requirement is that you have setup your [Lupusec component](/components/lupusec/).
+
+To use Lupusec devices in your installation, add the following `lupusec` section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+lupusec:
+      username: user 
+      password: 'greatpassword'
```
Comment (frenck) [inline source/_components/alarm_control_panel.lupusec.markdown]: Please use all CAPS constant like placeholders. For example, `YOUR_PASSWORD`.

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+title: "Lupusec Alarm Control Panel"
+description: "Instructions on how to setup the Lupusec Alarm control panel within Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_category: Alarm
+ha_release: 0.79
```
Comment (frenck) [inline source/_components/alarm_control_panel.lupusec.markdown]: `0.79` has been shipped already some time ago. The next possible release would be `0.81`.

File: source/_components/binary_sensor.lupusec.markdown [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+layout: page
+title: "Lupusec Binary Sensor"
+description: "Instructions on how to integrate Lupusec binary sensors into Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_release: 0.79
```
Comment (frenck) [inline source/_components/binary_sensor.lupusec.markdown]: `0.79` has been shipped already some time ago. The next possible release would be `0.81`.

File: source/_components/lupusec.markdown [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+title: "Lupus Electronics Home Security"
+description: "Instructions on integrating Lupusec home security with Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_category: Hub
+ha_release: 0.79
```
Comment (frenck) [inline source/_components/lupusec.markdown]: `0.79` has been shipped already some time ago. The next possible release would be `0.81`.

File: source/_components/switch.lupusec.markdown [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+layout: page
+title: "Lupusec Switch"
+description: "Instructions on how to integrate Lupusec switches into Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_release: 0.79
```
Comment (frenck) [inline source/_components/switch.lupusec.markdown]: `0.79` has been shipped already some time ago. The next possible release would be `0.81`.

**Review by frenck — CHANGES_REQUESTED** [review]
Hi there @majuss, thank you for this PR.

I and @klaasnicolaas left you some comments. Could you please take a look? 👍

**Comment by majuss** [comment]
@frenck @klaasnicolaas Thanks for the comments! I will work them off.

**Review by frenck — DISMISSED** [review]

**Comment by frenck** [comment]
Thanks, @majuss.

Can be merged as soon as the parent PR gets merged.

**Comment by majuss** [comment]
Thank you for the fine tuning!

**Comment by frenck** [comment]
@majuss Your welcome sir! 👍

File: source/_components/lupusec.markdown [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+# Example configuration.yaml entry
+lupusec:
+      username: YOUR_USERNAME
+      password: YOUR_PASSWORD
+      ip_address: 192.168.178.35
+```
+
+{% configuration %}
+username:
+  description: The login username of your lupusec alarm panel.
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: Please write Lupusec with a capital letter here,

File: source/_components/lupusec.markdown [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  required: true
+  type: string
+password:
+  description: The login password of your lupusec alarm panel.
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: here,

File: source/_components/lupusec.markdown [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  required: true
+  type: string
+ip_address:
+  description: The IP-address of your lupusec alarm panel.
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: and here as well.

File: source/_components/lupusec.markdown [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+  required: true
+  type: string
+scan_interval:
+  description: Scanning interval in seconds. Lower that value to get faster responses from your alarm panel but raise the temperatures and ressource usage.
+  required: false
+  type: integer
+  default: 2
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: Isn't the default value 5 seconds? https://github.com/home-assistant/home-assistant/pull/17237/files#diff-e34648182c47cb2ba95be679ce4023b1R23
(I do see that in other places it's 2 so I'm not sure)

File: source/_components/lupusec.markdown [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+password:
+  description: The login password of your lupusec alarm panel.
+  required: true
+  type: string
+ip_address:
+  description: The IP-address of your lupusec alarm panel.
+  required: true
+  type: string
+scan_interval:
+  description: Scanning interval in seconds. Lower that value to get faster responses from your alarm panel but raise the temperatures and ressource usage.
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: Typo: ressource --> resource (and sentence seems a bit off)

File: source/_components/lupusec.markdown [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  required: false
+  type: integer
+  default: 2
+
```
Comment (tjorim) [inline source/_components/lupusec.markdown]: Remove this empty line

**Review by tjorim — COMMENTED** [review]
Good luck with your first PR, @majuss!
I left some feedback on the documentation.

**Review by tjorim — COMMENTED** [review]
Is "Cloud Push" the correct IoT class? Not Local?

**Comment by majuss** [comment]
@tjorim I think I got everything.

**Review by frenck — DISMISSED** [review]

**Comment by frenck** [comment]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+title: "Lupusec Alarm Control Panel"
+description: "Instructions on how to setup the Lupusec Alarm control panel within Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_category: Alarm
+ha_release: 0.81
```
Comment (MartinHjelmare) [inline source/_components/alarm_control_panel.lupusec.markdown]: 0.83

File: source/_components/alarm_control_panel.lupusec.markdown [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+ha_iot_class: "Local Push"
```
Comment (MartinHjelmare) [inline source/_components/alarm_control_panel.lupusec.markdown]: This is a polling integration.

File: source/_components/binary_sensor.lupusec.markdown [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+title: "Lupusec Binary Sensor"
+description: "Instructions on how to integrate Lupusec binary sensors into Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_category: Binary Sensor
+ha_release: 0.81
```
Comment (MartinHjelmare) [inline source/_components/binary_sensor.lupusec.markdown]: 0.83

File: source/_components/binary_sensor.lupusec.markdown [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+ha_iot_class: "Local Push"
```
Comment (MartinHjelmare) [inline source/_components/binary_sensor.lupusec.markdown]: See above.

File: source/_components/binary_sensor.lupusec.markdown [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+---
+
+The `LUPUS-Electronics` security control panel platform allows you to control your [Lupusec](https://www.lupus-electronics.de) devices.
+
+This component will add `door contacts` and `window contacts` to your home-assistant setup
+
+For the switches to work you have to setup your [Lupusec component](/components/lupusec/).
```
Comment (MartinHjelmare) [inline source/_components/binary_sensor.lupusec.markdown]: It's confusing to call them switches. Switches in home assistant are actuator devices. The page is for binary sensors.

File: source/_components/lupusec.markdown [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+title: "Lupus Electronics Home Security"
+description: "Instructions on integrating Lupusec home security with Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_category: Hub
+ha_release: 0.81
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: 0.83

File: source/_components/lupusec.markdown [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+ha_iot_class: "Local Push"
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: See above.

File: source/_components/lupusec.markdown [hunk 22]
```diff
@@ -0,0 +1,52 @@
...
+---
+
+The `Lupusec` component allow the user to integrate their Lupusec alarm control panel and ultimately all connected sensors and other devices. For more information about the LUPUS-Electronics security system please visit their [website](https://www.lupus-electronics.de).
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: ... allows the user...

File: source/_components/lupusec.markdown [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+
+The following devices are supported by the underlying `lupupy` python library and integrated into hass.
+
+- [Alarm Control Panel](/components/alarm_control_panel.lupusec/): Displays the alarm status and controls arming, disarming and home modus.
+- [Binary Sensor](/components/binary_sensor.lupusec/): Reports on `Quick Actions`, `Door Contacts`, `Connectivity` sensors (remotes, keypads, and status indicators), `Moisture` sensors, and `Motion` or `Occupancy` sensors.
+- [Switch](/components/switch.lupusec/): Reports on `Camera` devices and will download and show the latest captured still image.
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: There's no downloading or showing the latest image in home assistant?

File: source/_components/lupusec.markdown [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+
+To use Lupusec devices in your installation, add the following `lupusec` section to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+lupusec:
+      username: YOUR_USERNAME
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: Use two space indentation.

File: source/_components/lupusec.markdown [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+  type: string
+password:
+  description: The login password of your Lupusec alarm panel.
+  required: true
+  type: string
+ip_address:
+  description: The IP-address of your Lupusec alarm panel.
+  required: true
+  type: string
+scan_interval:
```
Comment (MartinHjelmare) [inline source/_components/lupusec.markdown]: This isn't configurable.

File: source/_components/switch.lupusec.markdown [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+layout: page
+title: "Lupusec Switch"
+description: "Instructions on how to integrate Lupusec switches into Home Assistant."
+date: 2018-10-03 16:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: lupusec.jpg
+ha_release: 0.81
```
Comment (MartinHjelmare) [inline source/_components/switch.lupusec.markdown]: 0.83

File: source/_components/switch.lupusec.markdown [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+ha_category: Switch
+ha_iot_class: "Local Push"
```
Comment (MartinHjelmare) [inline source/_components/switch.lupusec.markdown]: See above.

File: source/_components/lupusec.markdown (hunk 25 above)
Comment (MartinHjelmare) [reply source/_components/lupusec.markdown]: `name` config option is missing.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

**Comment by majuss** [comment]
I will work on it asap :) Thanks again @MartinHjelmare for the very helpful comments!

**Review by MartinHjelmare — APPROVED** [review]
Looks good! Can be merged when build passes.

**Pull request merged**